docs/core/event_handler/bedrock_agents.md

```diff
@@ -0,0 +1,171 @@
+
+!!! info "This is not necessary if you're installing Powertools for AWS Lambda (Python) via [Lambda Layer/SAR](../../index.md#lambda-layer){target="_blank"}."
+
+You need to add `pydantic` as a dependency in your preferred tool _e.g., requirements.txt, pyproject.toml_.
+
+As of now, both Pydantic V1 and V2 are supported. For a future major version, we will only support Pydantic V2.
+
+### Your first Agent
+
+To create a Bedrock Agent, use the `BedrockAgentResolver` to annotate your actions. This is
+similar to the way [all the other Powertools](api_gateway.md) resolvers work.
+
+Be aware that it's important to include a description for each API endpoint. The description is essential because it provides Bedrock Agents with an understanding of the primary function of your API action.
+
+=== "Lambda handler"
+
+    ```python hl_lines="4 9 12 21"
+    --8<-- "examples/event_handler_bedrock_agents/src/getting_started.py"
+    ```
+
+	1. `description` is a required field in order for Bedrock Agents to work.
+
+=== "Input payload"
+
+	```json hl_lines="7 9 16"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started.json"
+	```
+
+=== "Output payload"
+
+	```json hl_lines="12-14"
+	--8<-- "examples/event_handler_bedrock_agents/src/getting_started_output.json"
+	```
+
+The resolvers utilized by Bedrock Agents are also compatible with the full suite of Powertools utilities. This ensures seamless integration and functionality across the different tools provided by Powertools when working with Bedrock Agents.
+
+### Generating OpenAPI schemas
+
+To create a schema for your API, use the `get_openapi_json_schema` function provided by the Bedrock Agent resolver. This function will produce a JSON-serialized string that represents your schema. You have the option to either display this string output or save it to a file for future reference.
+
+=== "Generating the OpenAPI schem"
+
+    ```python hl_lines="24 25"
+    --8<-- "examples/event_handler_bedrock_agents/src/generating_openapi_schema.py"
+    ```
+
+=== "OpenAPI schema"
+
+    ```json hl_lines="13 16 24"
+    --8<-- "examples/event_handler_bedrock_agents/src/generating_openapi_schema.json"
+    ```
+
+To obtain the OpenAPI schema, execute the Python script from the command line interface (CLI). Upon execution, the script will generate the schema, which will be output directly to the console.
+
+### Creating your Agent in the AWS Console
+
+To create a Bedrock Agent, you should refer to the [official documentation](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-create.html) provided by AWS.
+
+During the creation process, when the user interface (UI) prompts you for an OpenAPI specification, you should input the specification that was generated in the previous step. This is the spec that you obtained by running your Python script, which produced the OpenAPI schema as output.
+
+### Enabling the Swagger UI
+
+--8<-- "docs/core/event_handler/_swagger_ui.md"
```

Comment:
This line includes:
>Behind the scenes, the [data validation](http://127.0.0.1:8000/core/event_handler/api_gateway/#data-validation) feature auto-generates an OpenAPI specification from your routes and type annotations. You can use [Swagger UI](https://swagger.io/tools/swagger-ui/) to visualize and interact with your newly auto-documented API.

I think it's hard to understand what the documentation is trying to say here. Customers move from "Creating your Agent in the AWS Console" to Swagger data validation. We need to rethink this part of the documentation.